core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/container/postgres/PostgresConfigService.java

```diff
@@ -56,10 +64,25 @@ public void decorateServicePillarWithPostgresIfNeeded(Map<String, SaltPillarProp
         Map<String, Object> postgresConfig = new HashMap<>();
         if (dbServerConfigurer.isRemoteDatabaseNeeded(cluster)) {
             postgresConfig.put("configure_remote_db", "true");
+        } else {
+            collectEmbeddedDatabaseConfigs(stack, postgresConfig);
         }
         rdsConfigProviderFactory.getAllSupportedRdsConfigProviders().forEach(provider ->
                 postgresConfig.putAll(provider.createServicePillarConfigMapIfNeeded(stack, cluster)));
         return postgresConfig;
     }
 
+    private void collectEmbeddedDatabaseConfigs(Stack stack, Map<String, Object> postgresConfig) {
```

Comment:
I made some suggestion on slack for refactoring this method

Reply:
Refactored based on what we discussed on slack